setup.py

```diff
@@ -7,50 +7,19 @@
     from distutils.core import setup
 import ast
 
-
+current_dir = os.path.dirname(__file__)
 def read(fname):
-    inf = open(os.path.join(os.path.dirname(__file__), fname))
+    inf = open(os.path.join(current_dir, fname), "rt", encoding="utf-8")
```

Comment:
`rt` are default for `open`, so IMHO there won't be any gain being that explicit here.  Additionally, if this gets merge,, the only call for `read` is to get the version, thus the code below could be refactored to

```python3
with open(os.path.join(op.path.dirname(__file__), "appdirs.py")) as source:
    version = next(ast.literal_eval(line.split("=", 1)[1].strip())
                   for line in source if line.startswith("__version__"))
```